mage_ai/io/qdrant.py

```diff
@@ -0,0 +1,177 @@
+            collection_name (str): name of the collection.
+            Defaults to the name defined in io_config.yaml.
+        Returns:
+            collection created.
+        """
+        collection_name = self.collection if collection_name is None else collection_name
+        distance = models.Distance.COSINE if distance is None else distance
+        return self.client.create_collection(
+            collection_name=collection_name,
+            vectors_config=models.VectorParams(
+                size=vector_size,
+                distance=distance),
+        )
+
+    def load(
+        self,
+        limit_results: int,
+        query_vector: List,
+        collection_name: str = None,
+        **kwargs,
+    ) -> DataFrame:
+        """
+        Loads the data from Qdrant with query_vector.
+        Args:
+            limit_results (int): Number of results to return.
+            query_vector (List): vector list used to query.
+            collection_name (str): name of the collection.
+            Defaults to the name defined in io_config.yaml.
+        Returns:
+            DataFrame: Data frame object loaded with data from qdrant
+        """
+        # Assume collection is already created and exists.
+        collection_name = collection_name or self.collection
+
+        hitted_results = self.client.search(
+            collection_name=collection_name,
+            query_vector=query_vector,
+            limit=limit_results,
+            with_vectors=True,
+        )
+        self.client.close()
+
+        output_df = {}
+        output_df['id'] = [hit.id for hit in hitted_results]
+        output_df['payload'] = [hit.payload for hit in hitted_results]
+        output_df['score'] = [hit.score for hit in hitted_results]
+        output_df['vector'] = [hit.vector for hit in hitted_results]
+
+        return DataFrame.from_dict(output_df)
+
+    def export(
+        self,
+        df: DataFrame,
+        document_column: str,
+        id_column: str = None,
+        vector_column: str = None,
+        collection_name: str = None,
+        vector_size: int = None,
+        distance: models.Distance = None,
+        **kwargs,
+    ) -> None:
+        """
+        Save data into Qdrant.
+        Args:
+            df (DataFrame): Data to export.
+            document_column (str): Column name containinng documents to export.
+            id_column (str): Column name of the id. Default will use index in df.
+            vector_column (str): Column name of the vector. Will use default
+            encoder to auto generate query vector to auto generate query vector.
+            collection_name (str): name of the collection.
+            vector_size (int): dimension size of vector.
+            distance (models.Distance): distance metric to use.
+        """
+        collection_name = collection_name or self.collection
+        encoder = SentenceTransformer("all-MiniLM-L6-v2")
```

Comment:
can we make the "all-MiniLM-L6-v2" string a constant?

Reply:
also single quotes

Reply:
👍 Updated it to a constant.